tests/spark/transformations/test_http_request.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+
+import pytest
+
+from koheesio.spark import DataFrame, SparkSession  # type: ignore
+from koheesio.spark.transformations.http_request import HttpRequestTransformation  # type: ignore
+
+
+@pytest.fixture
+def input_df(spark: SparkSession) -> DataFrame:
+    """A simple DataFrame containing two URLs."""
+    return spark.createDataFrame(
+        [
+            (101, "http://www.textfiles.com/100/adventur.txt"),
+            (102, "http://www.textfiles.com/100/arttext.fun"),
+        ],
+        ["key", "url"],
+    )
+
+
+@pytest.fixture
+def download_path(tmp_path: Path) -> Path:
+    _path = tmp_path / "downloads"
+    _path.mkdir(exist_ok=True)
+    return _path
+
+
+class TestHttpRequestTransformation:
+    """
+    Input DataFrame:
+
+    | key | url                                        |
+    |-----|--------------------------------------------|
+    | 101 |	http://www.textfiles.com/100/adventur.txt  |
+    | 102 |	http://www.textfiles.com/100/arttext.fun   |
+
+    Output DataFrame:
+
+    | key | url                                        | downloaded_file_path  |
+    |-----|--------------------------------------------|-----------------------|
+    | 101 |	http://www.textfiles.com/100/adventur.txt  | downloads/adventur.txt|
+    | 102 |	http://www.textfiles.com/100/arttext.fun   | downloads/arttext.fun |
+
+    """
+
+    def test_downloading_files(self, input_df: DataFrame, download_path: Path) -> None:
+        """Test that the files are downloaded and the DataFrame is transformed correctly."""
+        # Arrange
+        expected_data = [
```

Comment:
I'm pretty sure I saved the expected data in a file already in the data folder of the tests right? Either way, I would like to take this hardcoding out of the code and refer to a file instead.

Reply:
Not 100% sure if it doesn't happen somewhere else, but the test_downloading_files is only checking the dataframe and if the files exist. not the contents of the downloaded file.

Reply:
I'm gonna put the content in test files btw. ;)